### ISSUE TITLE
Remove the deprecated scripts/ directory

### ISSUE THREAD

**Issue by benjaoming** [learningequality/ka-lite#3627]
Getting issues like #3624 should remind us that `scripts/` can be removed in 0.14

Any feelings about this?

I would suggest the following two replacements:

1. Functional installers + PyPi packages will take over the role of installing and removing KA Lite.
1. Documentation chapters handling how to add KA Lite to start up environments in case installers don't do that -- this is mainly relevant for Linux distributions, so it shouldn't be that long an example.

**Comment by aronasorman** [comment]
Yes, this is what we want. By release time our supported installers should be capable enough to handle all the necessary tasks (including uninstallation).

**Comment by benjaoming** [comment]
Would there be others interested in stepping up for this one?

**Comment by MCGallaspy** [comment]
@benjaoming can you outline what you have in mind for this:
>Functional installers + PyPi packages will take over the role of installing and removing KA Lite.

**Comment by benjaoming** [comment]
"Functional"... haha, I just meant that the installers (current Windows and Mac + new Debian) should be functional as in "not broken" (not functional programming) and we can assume that they will be once we release 0.14 :)

For other purposes of installation, we will also release on PyPi so you can do "pip install kalite"

**Comment by MCGallaspy** [comment]
Okay, I just wasn't clear about prior work on this -- the idea is that running setup.py should take care of all the setup stuff (including adding the kalite executable to the CL path) regardless of OS? And the current state is that there are issues on all OSes?

**Comment by benjaoming** [comment]
```
ls -1 scripts/
00 THIS DIRECTORY IS DEPRECATED
add_systemstart_task.bat  # New documentation entry!
createshortcut.vbs  # Installer does this
cronstart.bat  # Replaced by kalite command
cronstart.sh  # Replaced by kalite command
cronstop.bat  # Replaced by kalite command
cronstop.sh  # Replaced by kalite command
donotrunatboot.sh  # New documentation entry!
graph_models.sh  # Gah, don't know about this... maybe just get rid of it for now
optimizerpi.sh  # New documentation entry!
python.bat  # Replaced by kalite command
python.sh  # Replaced by kalite command
README.md  # Explains that the directory is deprecated :)
remove_systemstart_task.bat  # Installer does this... well actually, the uninstaller does it :)
runatboot.sh  # Replace with documenation entry (this is for init.d stuff on Linux)
run_command.bat  # kalite command does this
run_command.command  # kalite command does this
run_command.sh  # kalite command does this
runhidden.vbs  # Some weird utility script, nevermind
serverstart.bat  # Replaced by kalite command
serverstart.sh  # Replaced by kalite command
serverstop.bat  # Replaced by kalite command
serverstop.sh  # Replaced by kalite command
start.bat  # Replaced by kalite command
start.command  # Replaced by kalite command
start.sh  # Replaced by kalite command
stop.bat  # Replaced by kalite command
stop.command  # Replaced by kalite command
stop.sh  # Replaced by kalite command
uninstall.sh  # Handled by installers / pip
```

**Comment by benjaoming** [comment]
@MCGallaspy Not sure what the outcome will be for the Mac and Windows installers... for sure it will be good that in the future we have all installation going through setup.py, but for the 0.14 release, it'll be fine that everything regarding setup.py works on all platforms.. but the installers may still wish to distribute kalite in their own current fashion... they have some additional stuff about user interface, start up scripts etc...

But you can safely assume that the installers will install kalite properly and that this means everything in the scripts/ directory can be tossed.

**Comment by benjaoming** [comment]
Current state: Yes there are issues, maybe on all OSes... I'll make another announcement during the week... but I think that writing documentation about this is fine, you can just assume that `pip install ka-lite` will install it from PyPi and it will work after that :)

**Comment by aronasorman** [comment]
That would be cool if `pip install ka-lite` is one of our supported installations in the documentation. That would mean that we have a standard installation process for all other OSes out there :)

**Comment by benjaoming** [comment]
@cpauya and @ruimalheiro can you guys confirm if the Mac/Win installers are independent from these scripts now and if they need them, can you copy them out and maintain them together with the installers instead?

**Comment by MCGallaspy** [comment]
It appears the windows installer does indeed depend on some of these scripts. See [these search results](https://github.com/learningequality/installers/search?utf8=%E2%9C%93&q=bat). No word on mac, perhaps @cpauya can weigh in?

**Comment by cpauya** [comment]
We don't use those scripts on the Mac installer anymore since we have been using the `bin/kalite` command.

It doesn't do the following just yet:

1. use the `pip install ka-lite`
1. it doesn't have an uninstaller

**Comment by benjaoming** [comment]
@cpauya that's great!! Btw in case you use "pip" for installing, you can also use it for uninstalling.

@ruimalheiro could you phase out the scripts/ folder for the next release of the installer?

**Comment by MCGallaspy** [comment]
Punting to 0.15 as discussed in the standup.

**Comment by benjaoming** [comment]
I never really got around to comment on this decision.

Do we really wanna ship broken things that we aren't testing and already have deprecated?

AFAIK it's only the windows scripts that *might* be used and then a raspberry pi script.

**Comment by benjaoming** [comment]
Look at this...

https://github.com/learningequality/ka-lite/blob/0.14.x/scripts/cronstart.sh

It will start a cronserver that's already started by `kalite start`.

And look at this:

https://github.com/learningequality/ka-lite/blob/0.14.x/scripts/serverstart.sh

I'm not sure who depends on these scripts. If someone created an init.d script with them, the script will remain and call existing (but deprecated) management commands.

The only people I can think of whom it'd break for would be the ones who have put "start.sh" directly in their rc.local or similar individual configuration... but they would notice that the server doesn't start and the fix would be pretty simple.

I vote: Remove all linux/mac scripts, keep the optimizerpi.sh and the windows stuff... and then do a quick search through our documentation to see that we haven't put mentions of these scripts anywhere.

**Comment by MCGallaspy** [comment]
Obviously I am for removing these, but I think it's just too late for 0.14. If we let more PRs creep in we risk extending the debug time more and more.

**Comment by benjaoming** [comment]
We don't have to "debug" this, it doesn't affect the code base?

**Comment by MCGallaspy** [comment]
But it may affect the installers.

**Comment by MCGallaspy** [comment]
To clarify: I just wish to draw a line in the sand. While this is important, I do not think it is a show-stopper. We still have other issues which need to be resolved for 0.14 which we have identified as higher priorities, so let's focus on those. I disagree with your premise that this is a safe change, but only because I have insufficient information.

**Comment by benjaoming** [comment]
It's about leaving scripts that support a deprecated behavior and the chance that we just push the workload ahead of us that worries me here.

If the installers break, we can fix it. If users beyond the series of tubes experience breakage when they continue using these, we can't really fix it.

But of course, on the positive side.. the "scripts" dir isn't actually released on PyPi or in the PPA.. it's mainly here on Github.. and it already have a couple of warnings.

I'm gonna clean up the docs still in case they are mentioned.

**Comment by aronasorman** [comment]
@benjaoming can you try this out on develop once we've forked?

**Comment by MCGallaspy** [comment]
For windows, I'm pretty sure I got 'em all in https://github.com/learningequality/installers/pull/57. So I feel confident we can remove these scripts finally.

**Comment by aronasorman** [comment]
@MCGallaspy thanks, @benjaoming can you do the honors of deleting this folder?

**Comment by rtibbles** [comment]
Fixed in #4502
